core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -637,6 +688,89 @@ public void testParam() throws Exception {
         }
     }
 
+    @Test
+    public void testAdvancedAnnotatedHttpService() throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
+            HttpGet httpGet;
+            HttpPost httpPost;
+
+            try (CloseableHttpResponse res = hc.execute(new HttpGet(rule.httpUri("/8/same/path")))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET"));
```

Comment:
Could you update the test case so that we check the values of `content-type` headers in responses?